core/modules/wsched/SchedulerBase.cc

```diff
@@ -65,4 +65,74 @@ void SchedulerBase::setPriorityDefault() {
 }
 
 
+int SchedulerBase::_incrCountForUserQuery(uint64_t queryId) {
+    std::pair<uint64_t, int> entry(queryId, 0);
+    std::lock_guard<std::mutex> lock(_countsMutex);
+    auto result = _userQueryCounts.insert(entry);
+    return ++result.first->second;
```

Comment:
If `map[key]` has to add new value to the map it's guaranteed to be zero-initialized, one of the rare cases in C++ when behavior is intuitive :smile_cat: